Lib/TestRunnerBase.py

```diff
@@ -0,0 +1,338 @@
+        else:
+            self.ncpus = 1
+
+        if args.attributes:
+            self.nosetests_attr = args.attributes
+        else:
+            self.nosetests_attr = []
+
+        if get_sample_data == True:
+            download_sample_data_files(os.path.join(sys.prefix, "share", 
+                                                    self.test_suite_name, "test_data_files.txt"),
+                                       get_sampledata_path())
+            
+    def __is_option_set(self, option):
+        return self.run_options & option
+    
+    def __get_tests(self, failed_only=False, tests=None):
+        """
+        get_tests() gets the list of test names to run.
+        If <failed_only> is False, returns the set of the specified
+        test names. If <tests> is not specified, returns a set of all tests
+        under 'tests' sub directory.
+        If <failed_only> is True, returns the set of failed test names
+        from previous run.
+        If <failed_only> is True, and <tests> is specified, returns 
+        the set of failed test names that is listed in <tests>
+        """
+        print("xxx xxx in __get_tests...")
+        if tests is None:
+            # run all tests
+            print("xxx xxx glob")
+            test_names = glob.glob("tests/test_*py")
+            print("xxx xxx test_names...1...: {t}".format(t=test_names))
+        else:
+            test_names = set(tests)
+            print("xxx xxx test_names...2...: {t}".format(t=test_names))
+
+        if failed_only and os.path.exists(os.path.join("tests",".last_failure")):
+            f = open(os.path.join("tests", ".last_failure"))
+            failed = set(eval(f.read().strip()))
+            f.close()
+            new_tests = []
+            for failed_test in failed:
+                if failed_test in test_names:
+                    new_tests.append(failed_test)
+            test_names = new_tests
+
+        return(test_names)
+
+
+    def __get_baseline(self, workdir):
+        """
+        __get_baseline(self, workdir):
+        <workdir> : should be repo dir of the test
+        """    
+        os.chdir(workdir)
+        ret_code, cmd_output = self.__run_cmd('git rev-parse --abbrev-ref HEAD')
+        o = "".join(cmd_output)
+        branch = o.strip()
+        print("xxx xxx DEBUG DEBUG...{b}".format(b=branch))
+        if not os.path.exists("uvcdat-testdata"):
+            cmd = "git clone git://github.com/cdat/uvcdat-testdata"
+            ret_code, cmd_output = self.__run_cmd(cmd)
+            if ret_code != SUCCESS:
+                return ret_code
+        os.chdir("uvcdat-testdata")
+        ret_code, cmd_output = self.__run_cmd("git pull")
+        if ret_code != SUCCESS:
+            return ret_code
+        print("BRANCH WE ARE TRYING TO CHECKOUT is (%s)" % branch)
+        ret_code, cmd_output = self.__run_cmd("git checkout %s" % (branch))
+        os.chdir(workdir)
+        return(ret_code)
+
+    def __install_vtk_cdat(self):
```

Comment:
I wonder if we should tweak this to let it install any package name from a specific chanel and version?

Reply:
I removed this function as I am not sure if I need it.